### PR TITLE
hoc2015: incorporate another round of feedback on level design.

### PR DESCRIPTION
**Pull request by breville**
Just in time for loc pass beginning tomorrow.

### CODE REVIEW

File: apps/src/studio/levels.js [hunk 1]
```diff
@@ -1923,9 +1923,9 @@ levels.js_hoc2015_move_diagonal = {
      [0x00, 0x0000000, 0x0000000, 0x0000000, 0x0010000, 0x0000010, 0x0000000, 0x00],
      [0x00, 0x1100000, 0x1100000, 0x0000000, 0x0000001, 0x0000000, 0x0000000, 0x00],
      [0x00, 0x1100000, 0x1100000, 0x0000001, 0x0240000, 0x0250000, 0x0000000, 0x00],
-     [0x00, 0x0000000, 0x0000001, 0x0000000, 0x0000000, 0x0000000, 0x0000000, 0x00],
+     [0x00, 0x0000020, 0x0000001, 0x0000000, 0x0000000, 0x0000000, 0x0000000, 0x00],
      [0x00, 0x0000000, 0x0000000, 0x0000000, 0x0000000, 0x0000000, 0x0000000, 0x00],
-     [0x00, 0x0000000, 0x0000000, 0x1340000, 0x1340000, 0x1350000, 0x1350000, 0x20],
+     [0x00, 0x0000000, 0x0000000, 0x1340000, 0x1340000, 0x1350000, 0x1350000, 0x00],
```
Comment (islemaster) [inline apps/src/studio/levels.js:1928]: Moving the guy back to the left?

File: apps/src/studio/levels.js [hunk 2]
```diff
@@ -2044,7 +2044,7 @@ levels.js_hoc2015_move_around = {
      [0x0000000, 0x0000000, 0x00, 0x0000000, 0x0000000, 0x0000000, 0x0000000, 0x00],
      [0x0000000, 0x0000000, 0x00, 0x0000010, 0x0000000, 0x0000001, 0x0010000, 0x00],
      [0x0000000, 0x0000000, 0x00, 0x0040000, 0x0020000, 0x0000000, 0x0000000, 0x00],
-     [0x0000000, 0x0000000, 0x20, 0x0140000, 0x0000000, 0x0000001, 0x0000000, 0x00],
+     [0x0000000, 0x0000000, 0x20, 0x0140000, 0x0000000, 0x0000001, 0x0000000, 0x20],
```
Comment (islemaster) [inline apps/src/studio/levels.js:2047]: And adding another guy here, I think?

File: apps/src/studio/levels.js [hunk 3]
```diff
@@ -2489,6 +2522,8 @@ levels.js_hoc2015_win_lose = {
         blocklyMessage: msg.failedWinLoseGoalsBlockly()
       }
     },
+
+    // Success: enough itmes, enough points
```
Comment (islemaster) [inline apps/src/studio/levels.js]: s/itmes/items

File: apps/src/studio/studio.js [hunk 4]
```diff
@@ -5277,7 +5295,7 @@ Studio.conditionSatisfied = function(required) {
     var valueName = valueNames[k];
     var value = required[valueName];
 
-    if (valueName === 'timedOut' && Studio.timedOut() != value) {
+    if (valueName === 'timedOut' && tracked.timedOut  != value) {
```
Comment (islemaster) [inline apps/src/studio/studio.js]: Extra space?

File: apps/src/studio/studio.js [hunk 5]
```diff
@@ -5393,9 +5411,14 @@ var checkFinished = function () {
...
-  if (Studio.timedOut()) {
-    Studio.result = ResultType.FAILURE;
-    return true;
+  // Don't process timedOut condition here if we have progressConditions to take care of
+  // things, which can include a timedOut.  This avoids having this condition kick in earlier
+  // than level.progressConditions can take care of a timedOut.
+  if (!level.progressConditions) {
+    if (Studio.timedOut()) {
+      Studio.result = ResultType.FAILURE;
+      return true;
```
Comment (islemaster) [inline apps/src/studio/studio.js:5420]: Should this then be using `tracked.timedOut` instead of `Studio.timedOut()`?

**Comment by islemaster** [comment]
Interesting change with where we check conditions.  Seems good though.  LGTM!

File: apps/src/studio/studio.js (hunk 5 above)
Comment (breville) [inline apps/src/studio/studio.js:5420]: Good question.  In the interest of maintaining legacy compat, I'm going to leave as-is.  tracked.timeOut is set a bit later, in a manner that's compatible with the progressConditions system, but this check still operates on non-progressConditions levels.

**Pull request merged**